src/jllvm/class/ByteCodeIterator.hpp

```diff
@@ -257,6 +259,16 @@ struct ArrayOp : ByteCodeBase
         TLong = 11
     };
 
+    struct ArrayInfo
+    {
+        llvm::StringRef descriptor;
+        llvm::Type* type{};
+        std::size_t size{};
+        std::size_t elementOffset{};
+    };
+
+    ArrayInfo resolve(llvm::IRBuilder<>&);
```

Comment:
I think this method here is a bit of a layering violation. So far, all things in the `class` directory is about consuming class files, adding this method here which now also pulls in aspects of the object representation and LLVM compilation is odd imo. I don't really see a lot of reuse of it either.

I think it'd be better as a function within `ByteCodeCompileLayer.cpp` for that reason